sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -1041,6 +1054,23 @@ _detail::WinHttpRequest::~WinHttpRequest()
         Logger::Level::Informational,
         "WinHttpRequest::~WinHttpRequest. Closing handle synchronously.");
 
+    // Attempt to unregister the status callback. 
+    // if an operation was started the callback will be called regardless, 
+    // as it was initiated before we make the call to unregister
```

Comment:
I'm confused. How does the close handle event get registered since you unregistered the status callback and it is initiated *after* you unregistered the callback.

Reply:
updated comment